src/axis.js

```diff
@@ -472,8 +472,10 @@ c3_axis_fn.getId = function getId(id) {
     return id in config.data_axes ? config.data_axes[id] : 'y';
 };
 c3_axis_fn.getXAxisTickFormat = function getXAxisTickFormat() {
+    // #2251 previosly set any negative values to a whole number, 
+    // however both should be truncated according to the users format specification
     var $$ = this.owner, config = $$.config,
-        format = $$.isTimeSeries() ? $$.defaultAxisTimeFormat : $$.isCategorized() ? $$.categoryName : function (v) { return v < 0 ? v.toFixed(0) : v; };
+        format = $$.isTimeSeries() ? $$.defaultAxisTimeFormat : $$.isCategorized() ? $$.categoryName : function (v) { return v; };
```

Comment:
I'm really bored of those double ternary that are hard to read!

Why not:

```js
let format;
if ($$.isTimeSeries()) {
  format - $$.defaultAxisTimeFormat;
} else if ($$.isCategorized()) {
  format = $$.categoryName;
} else {
  format = (v) => v;
}
```
@kt3k I know this is not the current coding style of c3js but why make it hard on us understand/review ? What's your view on this?

Reply:
@panthony an alternative if you wanted to keep the ternary style but clear up the formatting could be:
```javascript
let format = ($$.isTimeSeries()) 
        ? $$.defaultAxisTimeFormat 
        : ($$.isCategorized()) 
            ? $$.categoryName 
            : function (v) { return v; };
```

Reply:
@muttox Or this way yes. That's way easier to read IMHO and at least when you modify a code path it does not highlight the whole thing.